app/matlab_reader.cpp

```diff
@@ -0,0 +1,452 @@
+#include "matlab_reader.h"
+#include "generic/read_csv.h"
+#include "generic/read_png.h"
+
+#include <unordered_map>
+#include <regex>
+#include <set>
+#include <fstream>
+#include <boost/log/trivial.hpp>
+#include <boost/range.hpp>
+
+namespace MouseTrack {
+
+/// Designed to capture upper and lower bounds for stream and frame numbers
+struct IndexAggregateSF {
+    StreamNumber firstStream = std::numeric_limits<FrameNumber>::max();
+    StreamNumber lastStream = -1;
+    FrameNumber firstFrame = std::numeric_limits<FrameNumber>::max();
+    FrameNumber lastFrame = -1;
+    std::set<std::string> suffixes;
+
+    IndexAggregateSF() :
+        firstStream(std::numeric_limits<StreamNumber>::max()),
+        lastStream(-1),
+        firstFrame(std::numeric_limits<FrameNumber>::max()),
+        lastFrame(-1)
+        {
+        // empty
+    }
+    void addStream(StreamNumber s){
+        firstStream = std::min(firstStream, s);
+        lastStream = std::max(lastStream, s);
+    }
+    void addFrame(FrameNumber f){
+        firstFrame = std::min(firstFrame, f);
+        lastFrame = std::max(lastFrame, f);
+    }
+    void addSuffix(std::string suff) {
+        suffixes.insert(suff);
+    }
+};
+
+
+/// Designed to capture upper and lower bounds for frame numbers
+struct IndexAggregateF {
+    FrameNumber firstFrame = std::numeric_limits<FrameNumber>::max();
+    FrameNumber lastFrame = -1;
+    std::set<std::string> suffixes;
+
+    IndexAggregateF() :
+        firstFrame(std::numeric_limits<FrameNumber>::max()),
+        lastFrame(-1)
+        {
+        // empty
+    }
+    void addFrame(FrameNumber f){
+        firstFrame = std::min(firstFrame, f);
+        lastFrame = std::max(lastFrame, f);
+    }
+    void addSuffix(std::string suff) {
+        suffixes.insert(suff);
+    }
+};
+
+// base file names for bag files
+const std::string NORMALIZED_DISPARITY_KEY{"depth_normalized"};
```

Comment:
() vs {} consistency

Reply:
[Prefer {} over () for initalization](https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives)

Reply:
done